metadata.md

```diff
@@ -0,0 +1,98 @@
+# Metadata Schema - Version 1.0
+
+In order to ensure interoperability between implementations of the Arrow in-memory format
+and associated storage formats such as Parquet and Feather, it is necessary to
+include additional top-level metadata.
+
+This information includes:
+
+-   spatial properties of each geometry column
+-   name of the primary geometry column
+
+Metadata is to be stored as a JSON-encoded UTF-8 string under the "geo" key within the
+top-level metadata object of the in-memory or file format.
+
+For clarity, the following shows the unencoded JSON structure:
+
+```python
+"geo": {
+        "columns": {...},  # REQUIRED: see below
+        "creator": {
+            "library": "<library name>",  # REQUIRED
+            "version": "<library version>",  # REQUIRED
+        },
+        "primary_column": "<primary geometry column name>",  # REQUIRED
+        "version": "1.0.0"  # REQUIRED
+    }
+```
+
+Each geometry column in the dataset must be included in the `columns` field above
+with the following content, keyed by the column name:
+
+```python
+"<column_name>": {
+    "bounds": [<xmin>, <ymin>, (zmin), <xmax>, <ymax>, (zmax)],  # OPTIONAL
+    "crs": "<WKT representation of CRS>",  # REQUIRED
```

Comment:
Do we want to make this required? (certainly best practice to always include, though)

Reply:
I think making it required helps reduce ambiguity:

if not present, it is either not set in the data source, or not included in the metadata (implementation error).  If it is present but set as `null`, then we can (theoretically) be more confident that the data source is lacking CRS information.

Reply:
Ah, yes, I was interpreting the "required" wrongly (I though of it as "needs to have a non-null value"). But I am certainly fine with requiring the key, but it might be good to mention somewhere that it can still be null (a reader of the metadata might need to be able to handle that).

Reply:
So no `EWKB`?

Reply:
@etiennebr is `EKWB` used heavily in R?  I see that it appears to be well-supported in `sf` particularly for I/O with postgis, but wasn't sure about how often it is used there.

On the Python side, my impression is that it is less well-used compared to WKB with CRS managed separately.  The implementations of to / from WKB were more easily at hand, and that drove some of the early use of `WKB` here.

Reply:
On the Python side, EWKB is also supported (although somewhat hidden). It's probably not much used, also because shapely does not directly expose a way to have a SRID on the geometry objects. 
(eg in geopandas we use that to infer srid from data coming from postgis)

Now, personally, I would argue strongly for a "one CRS for the full column" rule, and not allow geometries with varying CRS in a *single* column.

Is EWKB more than just SRID addition? Eg Z and M coordinates, is that included in base WKB, or is that EWKB?

Reply:
👍 on only one CRS per column.

I think the details of EWKB are covered [here](https://github.com/postgis/postgis/blob/2.1.0/doc/ZMSgeoms.txt#L61-L66)

Specifically around the addition of M and ZM variants of geometry and SRID support.

I could have misinterpreted, but my understanding of SRID is that it must be a `uint32` but it's interpretation (mapping to a concrete CRS) is subject to a list of SRIDs (e.g., [in postgis](https://postgis.net/docs/using_postgis_dbmanagement.html#spatial_ref_sys)), rather than necessarily following a given authority list (e.g., EPSG).  That the SRIDs are the same is by convention ("should") rather than prescription ("must"), right?

If we did add support for EWKB and did allow SRIDs within them, it may simply need to be up to the user to set and interpret those values correctly - which seems like it would impede portability between languages / libraries.

Reply:
`sf` only supports a single projection per column, so it would be compatible (it can still read it from `EWKB` if all projections are the same). I was thinking about other formats, such as postgis, but maybe it's out of scope.